### PR TITLE
Add comment editor

### PR DESCRIPTION
**Pull request by alshakero**
## Proposed Changes

This adds an asynchronously loaded block editor to the comment editing UI in Calypso. Which can be seen under `/comments/all`

It's added under a package called `verbum-block-editor`. With the goal being to publish it on NPM and use it in Verbum itself. 

## Testing Instructions
- Post a block comment on a site you own. Please try to use as many blocks as possible.
- Using the live link, please visit /comments/all.
- View the comment you posted, edit it and try you very best to break it.
- Save the comment. View it on the frontend. 
- Refresh the page and re-edit it. This makes sure the blocks are fine after editing.

### CODE REVIEW

File: client/my-sites/comments/comment/comment-edit.jsx [hunk 1]
```diff
@@ -198,11 +206,19 @@ export class CommentEdit extends Component {
...
 					</FormFieldset>
-
-					<CommentHtmlEditor
-						commentContent={ commentContent }
-						onChange={ this.setCommentContentValue }
-					/>
+					{ hasBlocks( commentContent ) ? (
+						<AsyncLoad
+							require="./comment-block-editor"
+							fallback={ <Spinner /> }
```
Comment (heavyweight) [inline client/my-sites/comments/comment/comment-edit.jsx]: We have `placeholder` and `fallback` that we can use. Want to double-check this later.

File: package.json [hunk 2]
```diff
@@ -171,16 +172,23 @@
...
-		"@wordpress/commands": "^0.19.0",
-		"@wordpress/components": "^25.14.0",
-		"@wordpress/data": "^9.18.0",
-		"@wordpress/element": "^5.25.0",
-		"@wordpress/i18n": "^4.48.0",
-		"@wordpress/icons": "^9.39.0",
-		"@wordpress/is-shallow-equal": "^4.48.0",
-		"@wordpress/primitives": "^3.46.0",
-		"@wordpress/url": "^3.49.0",
+		"@types/wordpress__block-editor": "^11.5.8",
```
Comment (heavyweight) [inline package.json:175]: Let's revisit these changes here later.

File: packages/verbum-block-editor/README.md
Comment (heavyweight) [inline packages/verbum-block-editor/README.md:1]: Maybe we add something here?

File: client/my-sites/comments/comment/comment-block-editor/index.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import {
+	Editor,
+	loadBlocksWithCustomizations,
+	loadTextFormatting,
+	addApiMiddleware,
+} from '@automattic/verbum-block-editor';
+import { parse } from '@wordpress/blocks';
```
Comment (heavyweight) [inline client/my-sites/comments/comment/comment-block-editor/index.tsx]: We should probably not depend on `@wordpress/blocks` here

File: packages/verbum-block-editor/src/editor/editor-settings.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Gutenberg Editor Settings
+ * @see https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/defaults.js
+ */
+export const editorSettings = {
+	disableCustomColors: false,
+	disableCustomFontSizes: false,
+	disablePostFormats: true,
+	isDistractionFree: false,
+	isRTL: false,
```
Comment (heavyweight) [inline packages/verbum-block-editor/src/editor/editor-settings.ts]: Let's make this locale based.

File: packages/verbum-block-editor/src/editor/editor-settings.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	autosaveInterval: 60,
+	codeEditingEnabled: false,
+	bodyPlaceholder: 'Leave a comment',
```
Comment (heavyweight) [inline packages/verbum-block-editor/src/editor/editor-settings.ts]: Let's make this locale-based.

File: packages/verbum-block-editor/src/editor/editor-settings.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	supportsLayout: false,
+	colors: [],
+	fontSizes: [],
+	imageDefaultSize: 'medium',
+	imageSizes: [],
+	imageEditing: false,
+	hasFixedToolbar: true,
+	maxWidth: 580,
```
Comment (heavyweight) [inline packages/verbum-block-editor/src/editor/editor-settings.ts]: What happens on bigger screens with this setting?

File: packages/verbum-block-editor/src/editor/inline-iframe-style.scss [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+		"Segoe UI",
+		Roboto,
+		Oxygen-Sans,
+		Ubuntu,
+		Cantarell,
+		"Helvetica Neue",
+		sans-serif;
+}
+
+/* This is need for useResizeObserver to work. */
```
Comment (heavyweight) [inline packages/verbum-block-editor/src/editor/inline-iframe-style.scss]:
```suggestion
/* This is needed for useResizeObserver to work. */
```

**Comment by agrullon95** [comment]
I added a comment with multiple blocks (image, youtube video, reblog post, and quote) , navigated to comments/all using the live link and saw these errors for all the blocks I added (comment editor is displaying broken blocks):

Im seeing this error for the image:

CleanShot 2024-01-04 at 02 04 51@2x

SImilar errors for youtube, and reblog blocks:

CleanShot 2024-01-04 at 02 07 05@2x



Error for the quote block:
CleanShot 2024-01-04 at 02 05 51@2x

**Comment by escapemanuele** [comment]
Incredible job @alshakero 

Tried to edit some text and had this
https://github.com/Automattic/wp-calypso/assets/52076348/68608a31-2f0b-4017-9832-a988e5ebf40d

**Review by heavyweight — CHANGES_REQUESTED** [review]
Posted some generic comments.
Will go into testing now

**Comment by renancarvalho** [comment]
I noticed that YouTube links are not visible on comments/all, but when clicking on edit I can see the embed code there.

image

image

**Comment by alshakero** [comment]
> Incredible job @alshakero
> 
> Tried to edit some text and had this [Automattic/wp-calypso/assets/52076348/68608a31-2f0b-4017-9832-a988e5ebf40d](https://github.com/Automattic/wp-calypso/assets/52076348/68608a31-2f0b-4017-9832-a988e5ebf40d)

Fixed!

File: client/my-sites/comments/comment/comment-edit.jsx (hunk 1 above)
Comment (alshakero) [reply client/my-sites/comments/comment/comment-edit.jsx]: Fixed!

File: client/my-sites/comments/comment/comment-block-editor/index.tsx (hunk 3 above)
Comment (alshakero) [reply client/my-sites/comments/comment/comment-block-editor/index.tsx]: Fixed.

File: packages/verbum-block-editor/src/editor/editor-settings.ts (hunk 5 above)
Comment (alshakero) [reply packages/verbum-block-editor/src/editor/editor-settings.ts]: Done
image

File: packages/verbum-block-editor/src/editor/editor-settings.ts (hunk 6 above)
Comment (alshakero) [reply packages/verbum-block-editor/src/editor/editor-settings.ts]: Nothing, all works well 
image

File: packages/verbum-block-editor/src/editor/inline-iframe-style.scss (hunk 7 above)
Comment (alshakero) [reply packages/verbum-block-editor/src/editor/inline-iframe-style.scss]: Fixed

**Comment by alshakero** [comment]
> I noticed that YouTube links are not visible on comments/all, but when clicking on edit I can see the embed code there.
> 
> image image

Yes! The block rendering is not yet done in Calypso. This is just for block editing.

**Review by heavyweight — APPROVED** [review]
Thanks for addressing the feedback and providing fixes so quickly!
Awesome job ❤️

**Pull request merged**